Strata/DDM/Format.lean

```diff
@@ -14,6 +14,57 @@ open Std (Format format)
 
 namespace Strata
 
+/--
+Check if a character is valid for starting a regular identifier.
+Regular identifiers must start with a letter or underscore.
+-/
+private def isIdBegin (c : Char) : Bool :=
+  c.isAlpha || c == '_'
+
+/--
+Check if a character is valid for continuing a regular identifier.
+Regular identifiers can contain letters, digits, underscores, and apostrophes.
+-/
+private def isIdContinue (c : Char) : Bool :=
+  c.isAlphanum || c == '_' || c == '\''
+
+/--
+Check if a string needs pipe delimiters when formatted as an identifier.
+Returns true if the string contains special characters, spaces, or starts with a digit.
+-/
+private def needsPipeDelimiters (s : String) : Bool :=
+  if s.isEmpty then
+    true
+  else
+    let chars := s.toList
+    !isIdBegin chars.head! || chars.any (fun c => !isIdContinue c)
```

Comment:
This causes the String to be converted into a List which has allocations and may not be cheap.

`s.toList.head!` can be converted into `s.startValidPos.get h` where `h` is a proof the string is non-empty. 

Also `s.toList.any` can be converted to `s.any`